### PR TITLE
Fix filtering by review state in aggregated list of analyses

### PR DESCRIPTION
**Pull request by juangallostra**
**Note**: This P.R. also removes some unnecessary code from `bika_listing.py`.

## Description of the issue/feature this PR addresses

1. Navigate to the aggregated list of analyses.
2. Try to filter by review state by clicking any of the buttons highlighted in the image below. The listing will not change and will always list the analyses associated to the first review state chosen.

seleccio_003

## Current behavior before PR

The aggregated analyses list does not get updated when changing the selected review state by clicking on it.

This is due to the fact that the ajax call done when clicking on a review state sends the current url. This url includes the query with the current review state (`http://localhost:8080/senaite/aggregatedanalyses?analyses_form_review_state=results_pending`). Hence processing the query will always point to the same list of analyses independently of the chosen review state. 

## Desired behavior after PR is merged

The aggregated list of analyses gets updated when changing the selected review state by clicking on it.

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/lims/browser/bika_listing.py [hunk 1]
```diff
@@ -588,13 +588,6 @@ def update(self):
         self.member = api.get_current_user()
         self.translate = self.context.translate
 
-        # Sanitize the request form
-        # XXX When does this happen?
-        if self.request.form:
-            for key, value in self.request.form.items():
-                if isinstance(value, list):
-                    self.request.form[key] = self.request.form[key][0]
-
```
Comment (ramonski) [inline bika/lims/browser/bika_listing.py]: Ok, we need to keep that one in mind. I wanted to delete it as well during refactoring, but decided to add only the comment. Definitely a weird piece of code and I have no clue when this happens.

File: bika/lims/browser/js/coffee/bika.lims.bikalisting.coffee [hunk 2]
```diff
@@ -380,6 +380,7 @@ class window.BikaListingTableView
     form_data.set "#{form_id}_review_state", state_id
 
     @ajax_submit
+      url: window.location.href.split("?")[0]
```
Comment (ramonski) [inline bika/lims/browser/js/coffee/bika.lims.bikalisting.coffee]: There is also a `get_base_url` method, that does exactly this;) Try to read first the code that exists, especially when it has been already refactored.

Comment (juangallostra) [reply bika/lims/browser/js/coffee/bika.lims.bikalisting.coffee]: @ramonski I see it now... Would you like me to change it? Sorry I missed that, I can update it without problems.

Comment (juangallostra) [reply bika/lims/browser/js/coffee/bika.lims.bikalisting.coffee]: Done in #718

**Pull request merged**